scripts/dev/build_release.py

```diff
@@ -99,6 +99,66 @@ def call_tox(
         [sys.executable, '-m', 'tox', '-vv', '-e', toxenv, *args],
         env=env, check=True)
 
+def fixup_qtwebengine():
+    qute_bundle = pathlib.Path('dist') / 'qutebrowser'
+    app_bundle = pathlib.Path('dist') / 'qutebrowser.app'
+
+    print(f"Attempting to fix up QtWebEngine in .app bundle: {app_bundle}")
+
+    top_level = app_bundle / "Contents" / "Resources"
+    qt_libdir = top_level / "PyQt6" / "Qt6" / 'lib' / 'QtWebEngineCore.framework' / 'Resources'
+
+    qute_lib_dir = qute_bundle / "PyQt6" / "Qt6" / 'lib' / 'QtWebEngineCore.framework' / 'Resources'
+
+    for resource in qt_libdir.iterdir():
+        if not resource.match('**/*pak'):
+            continue
+
+        name = resource.name
+        resource.unlink()
+        shutil.copy(qute_lib_dir / name, resource)
+
+    top_level = app_bundle / "Contents" / "MacOS"
+    qt_libdir = top_level / "PyQt6" / "Qt6" / "lib"
+
+    # For collected Qt .framework bundles, replace the copy of lib in top-level directory
+    # with symlink to the copy from the .framework bundle
+    for framework in qt_libdir.iterdir():
+        if not framework.is_dir():
+            continue
+        if not framework.match("**/*.framework"):
+            continue
+
+        # Infer shared lib name from .framework name
+        lib_name = framework.name[:-len('.framework')]
+        framework_lib_file = framework / "Versions" / "A" / lib_name
+
+        assert framework_lib_file.is_file(), "Framework lib file does not exist!"
+
+        # Check if library exists in top-level directory
+        lib_file = top_level / lib_name
+        if not lib_file.is_file():
+            continue
+
+        rel_framework_lib_file = framework_lib_file.relative_to(top_level)
+
+        print(f"Replacing {lib_file!s} with symbolic link to {framework_lib_file!s}; rel path: {rel_framework_lib_file}")
```

Comment:
how come these format specifiers have `!s` on them? That are just pathlib things like all the other ones right?